handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java

```diff
@@ -1470,12 +1470,15 @@ public synchronized void setSSLParameters(SSLParameters sslParameters) {
                     assert GET_ASCII_NAME_METHOD != null;
                     try {
                         List<?> servernames = (List<?>) GET_SERVER_NAMES_METHOD.invoke(sslParameters);
-                        for (Object serverName : servernames) {
-                            if (SNI_HOSTNAME_CLASS.isInstance(serverName)) {
-                                SSL.setTlsExtHostName(ssl, (String) GET_ASCII_NAME_METHOD.invoke(serverName));
-                            } else {
-                                throw new IllegalArgumentException("Only " + SNI_HOSTNAME_CLASS.getName()
-                                        + " instances are supported, but found: " + serverName);
+                        if (servernames != null) {
+                            for (Object serverName : servernames) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS134)